### PR TITLE
Users

### PR DESCRIPTION
**Pull request by anastasiiatulentseva**
Add basic user, configure devise and picture uploader

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+
+
```
Comment (stulentsev) [inline app/controllers/application_controller.rb]: too many blank lines

File: app/controllers/static_pages_controller.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 class StaticPagesController < ApplicationController
-  
+
   def home
+    @user = current_user
```
Comment (stulentsev) [inline app/controllers/static_pages_controller.rb]: a `before_action`, perhaps?

File: app/views/devise/passwords/edit.html.haml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  .row
+    .col-md-6.col-md-offset-2
+      = form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f|
+        = devise_error_messages!
+        = f.hidden_field :reset_password_token
+        .form-group
+          = f.label :password, "New password"
+          - if @minimum_password_length
+            %span.help-block
+              (#{@minimum_password_length} characters minimum)
```
Comment (stulentsev) [inline app/views/devise/passwords/edit.html.haml]: Hm, does this even work without `=` in front?

File: app/views/devise/registrations/edit.html.haml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f|
+  .row
+    .col-md-4
+      %div
+        %span
+          %strong Current picture:
+        %img.img-circle{:src => "#{@user.picture}"}/
+      .form-group
+        = f.label :picture, "Change picture:", class: "control-label"
+        = f.file_field :picture, accept: 'image/jpeg,image/gif,image/png'
```
Comment (stulentsev) [inline app/views/devise/registrations/edit.html.haml]: You have a constant for this. :)

File: config/environments/test.rb [hunk 5]
```diff
@@ -35,6 +35,8 @@
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
+  host = 'localhost:3000'
```
Comment (stulentsev) [inline config/environments/test.rb]: why the local var?

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = 'bb9fcc1867be8d3cc90b2fe812ecca71189046c59ce419f65f66774502dcb1ff6482ddaa39bd4b843b70ad98cab858ec0560e1bbb60778dfed47417afc1e2e71'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (stulentsev) [inline config/initializers/devise.rb:15]: You didn't change it

**Comment by stulentsev** [comment]
almost ok :)

File: app/views/devise/passwords/edit.html.haml (hunk 3 above)
Comment (anastasiiatulentseva) [inline app/views/devise/passwords/edit.html.haml]: this works somehow ) I'll fix

File: app/views/devise/registrations/edit.html.haml (hunk 4 above)
Comment (anastasiiatulentseva) [inline app/views/devise/registrations/edit.html.haml]: oops

File: config/initializers/devise.rb (hunk 6 above)
Comment (anastasiiatulentseva) [inline config/initializers/devise.rb:15]: I know that 
I don't know the correct name yet

File: app/views/devise/passwords/edit.html.haml (hunk 3 above)
Comment (stulentsev) [inline app/views/devise/passwords/edit.html.haml]: "if it ain't broken, don't fix it" :)

File: config/environments/test.rb (hunk 5 above)
Comment (anastasiiatulentseva) [inline config/environments/test.rb]: done

File: app/assets/javascripts/image_upload.coffee [hunk 7]
```diff
@@ -0,0 +1,6 @@
+do ->
+  $(document).on 'turbolinks:load', ->
+    $('.picture-size-check').bind 'change', ->
+      size_in_megabytes = @files[0].size / 1024 / 1024
+      if (size_in_megabytes > 5)
```
Comment (stulentsev) [inline app/assets/javascripts/image_upload.coffee]: parens are not needed, I think

File: app/controllers/static_pages_controller.rb [hunk 8]
```diff
@@ -1,10 +1,16 @@
 class StaticPagesController < ApplicationController
 
+  before_action :get_user
```
Comment (stulentsev) [inline app/controllers/static_pages_controller.rb]: More like "set_user" :)

File: app/views/devise/passwords/edit.html.haml [hunk 9]
```diff
@@ -9,7 +9,8 @@
           = f.label :password, "New password"
           - if @minimum_password_length
             %span.help-block
-              (#{@minimum_password_length} characters minimum)
+              =@minimum_password_length
```
Comment (stulentsev) [inline app/views/devise/passwords/edit.html.haml]: I like previous one better (if it works, that is)

**Pull request merged**